lib/galaxy/datatypes/binary.py

```diff
@@ -458,20 +458,40 @@ class Bam(BamNative):
     track_type = "ReadTrack"
     data_sources = {"data": "bai", "index": "bigwig"}
 
-    MetadataElement(name="bam_index", desc="BAM Index File", param=metadata.FileParameter, file_ext="bai", readonly=True, no_value=None, visible=False, optional=True)
+    MetadataElement(name="bam_index", desc="BAM Index File", param=metadata.FileParameter, file_ext="bam.bai", readonly=True, no_value=None, visible=False, optional=True)
```

Comment:
Is this safe to change?

Reply:
I don't think it is (even though it definitely is handy, but we can overwrite this elsewhere). 
```suggestion
    MetadataElement(name="bam_index", desc="BAM Index File", param=metadata.FileParameter, file_ext="bai", readonly=True, no_value=None, visible=False, optional=True)
```